### PR TITLE
feat(api): add bash script to make migrations

### LINKED ISSUE #123: [API] Add an sh script to generate migrations in correct order
In multi-app and multi-model django application, migrations are a way to keep track of db schema changes.

And our apps are dependent on each other and ideally should form a Directed Acyclic Graph of dependencies between them.

Everytime the develop needs to run makemigrations (be it first time setup or docker container setup), they should not have to worry about the order in which they have to `makemigrate` the apps.

A bash script that has this order defined will help the developers to skip this highly confusing step to build the app.

- Create a directory called `scripts` in the API `root`.
- Call this script `makemigrations.sh`
- It should be executable
- It should run makemigrations for all the apps that have models, in correct order.

### PR DESCRIPTION
**Pull request by arminpatel**
fixes #123

### CODE REVIEW

**Review by Aksh-Bansal-dev — COMMENTED** [review]
Make necessary changes in the Dockerfile.

**Comment by arminpatel** [comment]
> Make necessary changes in the Dockerfile.

done

File: api/scripts/makemigrations.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!bin/bash
```
Comment (Samy-33) [inline api/scripts/makemigrations.sh:1]: It should probably be `#!/bin/bash`

Try executing it: `./scripts/makemigrations.sh`

File: api/Dockerfile [hunk 2]
```diff
@@ -24,7 +24,7 @@ RUN pip install --upgrade pip && pip install -r requirements.txt
 COPY . $API_HOME  
 
 # make migrations
-RUN python manage.py makemigrations && python manage.py migrate
+RUN sh scripts/makemigrations.sh && python manage.py migrate
```
Comment (Samy-33) [inline api/Dockerfile:27]: `makemigrations.sh` should be executed independently by `./scripts/makemigrations`

That's how we are expecting it to be run in local and deployed env. The behaviour should be consistent.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

File: api/scripts/makemigrations.sh (hunk 1 above)
Comment (arminpatel) [reply api/scripts/makemigrations.sh:1]: > It should probably be `#!/bin/bash`
yeah 😅

File: api/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ Make sure you are using `python 3.8` as we are officially working on it. If some
 - Create a virtual environment: `$ python -m virtualenv .venv`
 - Activate the virtual environment: `$ source .venv/bin/activate` (On windows: `> ./.venv/Scripts/activate`)
 - Install the dependencies: `$ pip install -r requirements.txt`
-- Make Migrations: `$ python manage.py makemigrations` (This commands generates migrations which would be used to update database schemas)
+- Make Migrations: `$ sh scripts/makemigrations.sh` (This commands generates migrations which would be used to update database schemas)
```
Comment (Samy-33) [inline api/README.md:15]: Update the doc.

**Review by Samy-33 — CHANGES_REQUESTED** [review]
Make sure the script file is executable `chmod`
And the Dockerfile runs as expected.

**Comment by arminpatel** [comment]
Changed the script to execute independently and accordingly changed the doc. Docker image builds correctly.

**Review by Samy-33 — APPROVED** [review]
LGTM

**Pull request merged**